torch/_export/passes/replace_with_hop_pass_util.py

```diff
@@ -0,0 +1,99 @@
+# mypy: allow-untyped-defs
+
+import copy
+import operator
+from typing import Callable
+
+import torch
+from torch._ops import HigherOrderOperator
+
+from ..utils import node_replace_
+
+
+def _replace_with_hop_helper(
+    node: torch.fx.Node,
+    enter_block_node: torch.fx.Node,
+    node_filter: Callable,
```

Comment:
@yushangdi I'm wondering what is the purpose if that `node_filter`?

As far as I can see it is not used at all, making it a dead argument. Was the idea to use it eventually?

If not, should it be removed? It looks like the usages would become significantly simpler if they don't have to pass in that filter function.

Reply:
good catch, yeah I think it can be removed.

> @yushangdi I'm wondering what is the purpose if that `node_filter`?
> 
> As far as I can see it is not used at all, making it a dead argument. Was the idea to use it eventually?
> 
> If not, should it be removed? It looks like the usages would become significantly simpler if they don't have to pass in that filter function.

Reply:
PR for removing it: https://github.com/pytorch/pytorch/pull/141983